Export Analysis Reports/Sdl.Community.ExportAnalysisReports/Service/StudioService.cs

```diff
@@ -0,0 +1,244 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Sdl.Community.ExportAnalysisReports.Helpers;
+using Sdl.Community.ExportAnalysisReports.Interfaces;
+using Sdl.Community.ExportAnalysisReports.Model;
+using Sdl.Community.Toolkit.Core.Services;
+using Sdl.ProjectAutomation.Core;
+using Sdl.ProjectAutomation.FileBased;
+
+namespace Sdl.Community.ExportAnalysisReports.Service
+{
+	public class StudioService : IStudioService
+	{
+		public static readonly Log Log = Log.Instance;
+
+		public StudioService()
+		{
+			ProjectsXmlPath = GetStudioProjectsXmlPath();
+		}
+
+		public string ProjectsXmlPath { get; set; }
+		/// <summary>
+		/// Add file path names to list (used to identify when user opens a single file project)
+		/// </summary>
+		/// <param name="studioProjectsPaths"></param>
+		/// <returns></returns>
+		public List<string> AddFilePaths(List<string> studioProjectsPaths)
```

Comment:
Inversions and linq recommendations.

Reply:
Indeed the inversion and linq are now implemented. The changes will be reflected in the next commit